### PR TITLE
Fix unset singleton OwnedObject property

### LINKED ISSUE #269: AttributeError: 'NoneType' object has no attribute 'is_top_level'
Unsetting the `cd.sequence` results in this exception.

Example:
```
doc = sbol2.Document()
cd = doc.componentDefinitions.create('cd1')
cd.sequence = sbol2.Sequence('cd1_sequence')
cd.sequence = None
```

### PR DESCRIPTION
**Pull request by bbartley**
- Allow user to unset a singleton OwnedObject property by assigning None
- Add validation to `remove` methods
- Fix #269

### CODE REVIEW

File: sbol2/property.py [hunk 1]
```diff
@@ -761,6 +761,8 @@ def setSinglePropertyValue(self, new_value):
         # TODO: This can leave the attribute empty if `add` fails.
         # Can we capture that and sent the old value back again?
         self._sbol_owner.owned_objects[self._rdf_type].clear()
+        if new_value is None:
+            return
```
Comment (tcmitchell) [inline sbol2/property.py:765]: This will bypass the validation rules, which are fired at the end of the `add` method. Is that what we want? I feel like we want setting to `None` to still fire any associated validation rules. Do you agree? Or is that not necessary.

File: test/test_property.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import os
 
 import sbol2
+from sbol2.property import OwnedObject
```
Comment (tcmitchell) [inline test/test_property.py:5]: This seems like an unnecessary (and unused) import.

File: test/test_property.py [hunk 3]
```diff
@@ -151,6 +152,9 @@ def test_owned_object_singleton(self):
         cd.annotation = sbol.Identified('foo')
         self.assertEqual(type(cd.annotation), sbol.Identified)
 
+        # Test unsetting
+        cd.annotation = None
+
```
Comment (tcmitchell) [inline test/test_property.py:157]: We should probably add an assert to test that the value is actually set to `None` or some other null value.

File: sbol2/property.py (hunk 1 above)
Comment (bbartley) [reply sbol2/property.py:765]: Yeah the fix above is too simplistic.  In principle, it's probably a good idea to run validation rules when we unset or remove a value.  This probably boils down to seeing what happens with `self.validate(None)`. That might break some stuff.

Also instead of handling the unsetting by the `add` method, however, it would make more sense to handle the unsetting with the `remove` method.  But currently the `remove` methods don't currently invoke the validation rules.

File: test/test_property.py (hunk 3 above)
Comment (bbartley) [reply test/test_property.py]: Done

File: sbol2/property.py (hunk 1 above)
Comment (bbartley) [reply sbol2/property.py]: OK, 5ef42bb takes a different approach to unsetting
